zbus_macros/src/iface.rs

```diff
@@ -783,7 +783,11 @@ pub fn expand(args: Punctuated<Meta, Token![,]>, mut input: ItemImpl) -> syn::Re
                             #args_from_msg
                             let reply = self.#ident(#args_names)#method_await;
                             let hdr = message.header();
-                            #reply
+                            if hdr.primary().flags().contains(zbus::message::Flags::NoReplyExpected) {
+                                Ok(())
+                            } else {
+                                #reply
```

Comment:
hmm.. actually does it make any diff? You could explain a bit in the commit and PR description. :)